bin/cylc-cycle-point

```diff
@@ -266,9 +264,4 @@ def main():
 
 
 if __name__ == "__main__":
-    try:
-        main()
-    except Exception as exc:
-        if cylc.flags.debug:
-            raise
-        sys.exit(str(exc))
+    main()
```

Comment:
I guess I didn't review the original PR to master, but why this change? (My old "users should see an error message, not a traceback" argument is probably wrong, but we should be consistent, or remove this from all the commands, no?)

Reply:
No, your original assumption was not wrong. However:
* Time has changed and many users are now familiar enough with Python that the trace back is no longer scary - and it helps us and them when they send us bug reports.
* Catching known exception here is good, but catching *all* generic exceptions here is not so good - especially when you have those cryptic system errors - which is much better being a trace back.

You are right. I should changed everything.

Reply:
Yes, I'm happy to have this removed from all commands.

Reply:
##### Word of warning: In Python3 when you catch an exception its context is not forgotten which results in multi-part traceback which gets really messy really quick.

For example a pattern we use a lot in Cylc is to catch a nasty `OSError` or `KeyError` or whatever and raise a more user-friendly `CylcException`, `ConfigError` or whatnot. This pattern is at Python3 fundamentally broken as you will end up raising *both* the original exception *as well* as the new one.

For example say we are trying to catch `ZeroDivisionError` and hide it from the user replacing it with a more informative `Exception`...

```python
>>> try:
...     1/0
... except ZeroDivisionError:
...     # raise a completely different exception
...     raise Exception('foo')
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
ZeroDivisionError: division by zero

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
Exception: foo
```